### PR TITLE
Task list

### PR DESCRIPTION
**Pull request by Jazmusic1234**
Pulled latest version of develop branch and merged locally. Also fixed any ids that I missed the first time.

### CODE REVIEW

File: source/modules/task-list.js [hunk 1]
```diff
@@ -18,36 +32,53 @@
...
-export function saveTask() {
+function saveTask() {
     let taskNameInput = document.getElementById("task-name");
     let taskList = document.getElementById("task-list");
-    let newTask = createCustomTaskTag(taskNameInput.value);
-    taskList.appendChild(newTask);
+
+    if (SAVE_FLAG === SAVE_ON) { // Save new task
+        if (inputSanitizer(taskNameInput.value)){ // check for input
+            alert("Please enter something!!");
```
Comment (keskinmbaha) [inline source/modules/task-list.js:53]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
            alert('Please enter something!!');
```

File: source/modules/task-list.js [hunk 2]
```diff
@@ -1,15 +1,29 @@
...
  * @name addTaskButton
  * @function
  * @description Opens or closes add-task form
  */
-export function addTaskButton() {
+function addTaskButton() {
     let button = document.getElementById("add-task-form");
     if (button.classList.contains("hidden") === true) {
         button.classList.remove("hidden");
+        document.getElementById("task-name").focus();
```
Comment (keskinmbaha) [inline source/modules/task-list.js:23]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
        document.getElementById('task-name').focus();
```

File: source/main.js [hunk 3]
```diff
@@ -1,29 +1,39 @@
...
-// Task list
-document.getElementById('addTasksButton').addEventListener('click',addTaskButton);
-document.getElementById('saveButton').addEventListener('click',saveTask);
-document.getElementById('cancelButton').addEventListener('click',cancelTask);
 
-  
+// Task List
+document.getElementById("add-tasks-button").addEventListener("click", addTaskButton);
+document.getElementById("task-name").addEventListener("keypress", (event) => {
+    if (event.key === "Enter") // allow user to hit enter to save task
```
Comment (keskinmbaha) [inline source/main.js:32]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
    if (event.key === 'Enter') // allow user to hit enter to save task
```

File: source/main.js [hunk 4]
```diff
@@ -1,29 +1,39 @@
...
 // Timer
 document.getElementById('form-enabler').addEventListener('change', setCustomTime);
 document.getElementById('start-button').addEventListener('click', onStart);
 document.getElementById('reset-button').addEventListener('click', onReset);
 
 // Keyboard shortcuts
-document.addEventListener('keydown', keyboardShortcut);
+document.addEventListener('keydown', (event) => {
+    // only allow this event to be fired when task form is hidden
+    if (document.getElementById("add-task-form").classList.contains("hidden"))
```
Comment (keskinmbaha) [inline source/main.js:18]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
    if (document.getElementById('add-task-form').classList.contains("hidden"))
```

File: source/modules/task-list.js [hunk 5]
```diff
@@ -117,5 +203,38 @@
...
+}
+
+/**
+ * @name inputSanitizer
+ * @function
+ * @description helper function - set TASK_CONTENT for editing task
+ * @param content is the task container
+ */
+function inputSanitizer(input) {
+    if (input == "")
```
Comment (keskinmbaha) [inline source/modules/task-list.js:233]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
    if (input == '')
```

File: source/main.js [hunk 6]
```diff
@@ -1,29 +1,39 @@
...
 document.getElementById('close-settings').addEventListener('click', hideSettings);
 
-// Task list
-document.getElementById('addTasksButton').addEventListener('click',addTaskButton);
-document.getElementById('saveButton').addEventListener('click',saveTask);
-document.getElementById('cancelButton').addEventListener('click',cancelTask);
 
-  
+// Task List
+document.getElementById("add-tasks-button").addEventListener("click", addTaskButton);
```
Comment (keskinmbaha) [inline source/main.js:30]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
document.getElementById('add-tasks-button').addEventListener("click", addTaskButton);
```

File: source/main.js [hunk 7]
```diff
@@ -1,29 +1,39 @@
...
+document.getElementById("task-name").addEventListener("keypress", (event) => {
```
Comment (keskinmbaha) [inline source/main.js:31]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
document.getElementById('task-name').addEventListener("keypress", (event) => {
```

File: source/main.js (hunk 3 above)
Comment (keskinmbaha) [inline source/main.js:32]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
    if (event.key === 'Enter') // allow user to hit enter to save task
```

File: source/modules/task-list.js (hunk 1 above)
Comment (keskinmbaha) [inline source/modules/task-list.js:53]: Codacy has a fix for the issue: [Strings must use singlequote.](https://app.codacy.com/gh/cse112-sp22-group5/cse112-sp22-group5/pullRequest?prid=9414198)
```suggestion
            alert('Please enter something!!');
```

**Comment by EthanWadsworth** [comment]
Merging to add all features to current working version. Will need to go back and get code quality and styling up to standard. See failed builds on what to fix.

**Pull request merged**